utils/utils.go

```diff
@@ -196,19 +197,32 @@ func IsResponseEmptyOrMutiple(rLen int, resName string) error {
 }
 
 func CheckThrottling(httpResp *http.Response, err error) *retry.RetryError {
-	rand.Seed(time.Now().UnixNano())
+	rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
There is no need to seed the random number generator, you can remove this line. Since 1.20, it is already seeded with a random number, and your code does not care if your random numbers are really random...